staging/src/k8s.io/component-base/zpages/statusz/registry.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"time"
+
+	"github.com/blang/semver/v4"
+	"k8s.io/component-base/version"
+	"k8s.io/klog/v2"
+
+	"k8s.io/component-base/featuregate"
+	compbasemetrics "k8s.io/component-base/metrics"
+)
+
+type statuszRegistry interface {
+	processStartTime() time.Time
+	goVersion() string
+	binaryVersion() semver.Version
+	emulationVersion() semver.Version
+	minCompatibilityVersion() semver.Version
```

Comment:
are these defined for all components or only control plane components? will kubelet have `/statusz`?

Reply:
Only control plane components. Kubelet was also proposed to have a statusz in the KEP, yes.

Reply:
Kubelet's emulation and compat-version would point to the binary version by default I guess